ee/desktop/runner/server/server.go

```diff
@@ -65,28 +66,58 @@ func New(slogger *slog.Logger,
 
 	rs.slogger = slogger.With("component", "desktop_runner_root_server")
 
-	mux := http.NewServeMux()
+	authedMux := http.NewServeMux()
+	unauthedMux := http.NewServeMux()
 
 	// health check endpoint
-	mux.HandleFunc(HealthCheckEndpoint, func(w http.ResponseWriter, r *http.Request) {
+	authedMux.HandleFunc(HealthCheckEndpoint, func(w http.ResponseWriter, r *http.Request) {
 		if r.Body != nil {
 			r.Body.Close()
 		}
 	})
 
 	// menu opened endpoint
-	mux.HandleFunc(MenuOpenedEndpoint, func(w http.ResponseWriter, r *http.Request) {
+	authedMux.HandleFunc(MenuOpenedEndpoint, func(w http.ResponseWriter, r *http.Request) {
 		if r.Body != nil {
 			r.Body.Close()
 		}
 
 		controlRequestIntervalOverrider.SetControlRequestIntervalOverride(controlRequestAccelerationInterval, controlRequestAcclerationDuration)
 	})
 
-	mux.Handle(MessageEndpoint, http.HandlerFunc(rs.sendMessage))
+	authedMux.Handle(MessageEndpoint, http.HandlerFunc(rs.sendMessage))
+	authMuxHandler := rs.authMiddleware(authedMux)
+
+	unauthedMux.Handle(RecheckEndpoint, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Would we want to rate limit this endpoint to avoid kicking off a bunch of simultaneous rechecks? Found https://www.alexedwards.net/blog/how-to-rate-limit-http-requests for a fairly quick implementation, but also feeling like it's probably overkill.